include/dfrn.php

```diff
@@ -461,28 +487,31 @@ private static function add_header($doc, $owner, $authorelement, $alternatelink
 	 * @param string $authorelement Element name for the author
 	 *
 	 * @return object XML author object
+	 * @todo Add type-hints
 	 */
 	private static function add_author($doc, $owner, $authorelement, $public) {
 
 		// Is the profile hidden or shouldn't be published in the net? Then add the "hide" element
 		$r = q("SELECT `id` FROM `profile` INNER JOIN `user` ON `user`.`uid` = `profile`.`uid`
 				WHERE (`hidewall` OR NOT `net-publish`) AND `user`.`uid` = %d",
 			intval($owner['uid']));
-		if ($r)
+		if (dbm::is_result($r)) {
 			$hidewall = true;
-		else
+		} else {
 			$hidewall = false;
+		}
 
 		$author = $doc->createElement($authorelement);
 
 		$namdate = datetime_convert('UTC', 'UTC', $owner['name-date'].'+00:00', ATOM_TIME);
 		$uridate = datetime_convert('UTC', 'UTC', $owner['uri-date'].'+00:00', ATOM_TIME);
 		$picdate = datetime_convert('UTC', 'UTC', $owner['avatar-date'].'+00:00', ATOM_TIME);
 
-		if (!$public OR !$hidewall)
+		$attributes = array();
```

Comment:
With me it is fine and I do it this way:

1. Initialize variable with initial value (like `false` for booleans)
2. test conditions and set variable (e.g. returned value) to other value (like `true`) when conditions are met.

It is quite useless when it does nothing more than just initializing the variable with a default value and in `if()` block you do otherwise with it.